app/src/app/utils/events/mapEvents.ts

```diff
@@ -93,38 +92,47 @@ export const handleMapMouseDown = (
 export const handleMapMouseEnter = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
 ) => {};
 
 export const handleMapMouseOver = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
 ) => {};
 
 export const handleMapMouseLeave = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
-) => {};
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+  const sourceLayer = mapStore.selectedLayer?.name;
+  if (sourceLayer && hoverFeatureIds.current.size && activeTool === "brush") {
+    UnhighlightFeature(map, hoverFeatureIds, sourceLayer);
+  }
+};
 
 export const handleMapMouseOut = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
-) => {};
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  // console.log("mouse out");
+};
 
 export const handleMapMouseMove = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
 ) => {
   const mapStore = useMapStore.getState();
   const activeTool = mapStore.activeTool;
   const isPainting = mapStore.isPainting;
   const sourceLayer = mapStore.selectedLayer?.name;
   const selectedFeatures = mapStore.paintFunction(map, e, mapStore.brushSize);
-  if (!isPainting && sourceLayer) {
+  if (!isPainting && sourceLayer && activeTool === "brush") {
```

Comment:
H: This is super close but I think the behavior we want is:
```javascript
if (sourceLayer && activeTool === "brush") {
    HighlightFeature(selectedFeatures, map, hoverFeatureIds, sourceLayer);
    if (isPainting) {
      // selects in the map object; the store object
      // is updated in the mouseup event
      SelectMapFeatures(selectedFeatures, map, mapStore);
    }
}
```
Right now, when you're painting the highlighted features don't follow your cursor, but rather are shown where the mouse down event event occurred. Instead, we should continue to highlight features under the cursor as you paint (tested locally).

Reply:
good catch